src/peft/tuners/lora/bnb.py

```diff
@@ -233,7 +237,14 @@ def forward(self, x: torch.Tensor, *args, **kwargs) -> torch.Tensor:
                     if not self.use_dora[active_adapter]:
                         output = lora_B(lora_A(dropout(x))) * scaling
                     else:
-                        output = self._apply_dora(x, lora_A, lora_B, scaling, active_adapter)
+                        x = dropout(x)
```

Comment:
Ah yes, I wanted to mention this in the PR description but forgot. I think this was actually a bug in QDoRA previously, as dropout was not applied at all. For reference, see how we do it in normal LoRA:

https://github.com/huggingface/peft/blob/cb0bf077744d11524ec6f68d920f4cfe4ef3e8f3/src/peft/tuners/lora/layer.py#L564-L565

Since `dropout` is an instance of `nn.Dropout`, when we call `.eval()`, PyTorch will automatically deactivate dropout.

```python
>>> drop = nn.Dropout(0.5)
>>> x = torch.ones(10)
>>> drop(x)
tensor([2., 0., 0., 0., 2., 0., 0., 2., 2., 2.])
>>> drop.eval()
>>> drop(x)
tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
```

Reply:
Makes sense, thank you for explaining!